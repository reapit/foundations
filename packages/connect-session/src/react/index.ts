import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
