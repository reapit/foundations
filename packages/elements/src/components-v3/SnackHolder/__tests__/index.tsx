import * as React from 'react'
import { shallow } from 'enzyme'
import { SnackHolder } from '../'

describe('SnackHolder component', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(<SnackHolder snacks={[]} />)
    expect(wrapper).toMatchSnapshot()
  })
})
