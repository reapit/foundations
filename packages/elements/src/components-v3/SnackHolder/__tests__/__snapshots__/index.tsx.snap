// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackHolder component should match a snapshot 1`] = `
<ElSnackHolder>
  <Snack
    icon="info"
    intent="primary"
    onRemove={[Function]}
  >
    i am a snack
  </Snack>
</ElSnackHolder>
`;
